music21/musicxml/m21ToXml.py

```diff
@@ -3332,7 +3366,20 @@ def parseFlatElements(
         # That way chord symbols and other 0-width objects appear before notes as much as
         # possible.
         objIterator: OffsetIterator[base.Music21Object] = OffsetIterator(m)
-        for objGroup in objIterator:
+
+        # Prepare the iteration by offsets. If there are FiguredBassIndication objects
+        # first group them in one list together with their note, instead of handling them
+        # seperately. O
+        groupedObjList: list = []
+        for els in objIterator:
+            if len(groupedObjList) > 0:
+                if len(els) == 1 and isinstance(els[0], harmony.FiguredBassIndication):
+                    # print('**multiple fb found**')
+                    groupedObjList[-1].append(els[0])
+                    continue
+            groupedObjList.append(els)
```

Comment:
Did you explore solutions that did not involve another iteration of the entire stream?